### PR TITLE
Tweaks improving CI reliability

### PR DESCRIPTION
**Pull request by Romex91**
* Short summary: Print p-value for improvements and regressions
* fail when receiving bad HTTP statuses or console errors

### CODE REVIEW

File: packages/cli/src/compare/compare-results.ts [hunk 1]
```diff
@@ -118,6 +118,7 @@ export class CompareResults {
     this.phaseResultsFormatted.forEach((phaseData) => {
       const {
         phase,
+        pValue,
```
Comment (Romex91) [inline packages/cli/src/compare/compare-results.ts:121]: For better summary reports

File: packages/core/src/create-lighthouse-benchmark.ts [hunk 2]
```diff
@@ -31,6 +31,26 @@ async function runLighthouse(
     `${namePrefix}_performance_profile.json`,
     JSON.stringify(runnerResult.artifacts)
   );
+  if (runnerResult.lhr.runtimeError) {
```
Comment (Romex91) [inline packages/core/src/create-lighthouse-benchmark.ts:34]: Ignoring errors in perf tests is a reliable way to be confused. 
While working on [Popmenu CI perf tests](https://github.com/Popmenu/popmenu/pull/6480) I stumbled on some `unexpected improvements` that actually were `errors on the page` and spent some time scratching my head.
Failing CI jobs is much better.

**Comment by Romex91** [comment]
@justisb, I cannot add you as a reviewer for some reason. Could you change permissions for this repo?

**Comment by justisb** [comment]
@Romex91 I just added you as a maintainer so you should be able to merge/push updates.

**Pull request merged**